mlflow/deployments/base.py

```diff
@@ -87,7 +87,81 @@ def __init__(self, target_uri):
         self.target_uri = target_uri
 
     @abc.abstractmethod
-    def create_deployment(self, name, model_uri, flavor=None, config=None):
+    def create_endpoint(self, name, config=None):
+        """
```

Comment:
Instead of directly modifying `BaseDeploymentClient` class, which is meant to be used by all possible deployment plugins / implementations (some of which doesn't support the concept of endpoint), how about creating a new class deriving from `BaseDeploymentClient`? e.g. `BaseEndpointClient`, or `BaseDeploymentClientWithEndpoint`